contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1422 @@
+                tAmountToSlash,
+                100
+            );
+        }
+
+        // slash NU
+        if (tAmountToSlash > 0 && operatorStruct.nuInTStake > 0) {
+            // synchronization skipped due to impossibility of real discrepancy
+            tAmountToSlash = seizeNu(operatorStruct, tAmountToSlash, 100);
+        }
+
+        emit TokensSeized(
+            slashing.operator,
+            slashing.amount - tAmountToSlash,
+            false
+        );
+        authorizationDecrease(slashing.operator, operatorStruct);
+    }
+
+    /// @notice Synchronize authorizations (if needed) after slashing stake
+    function authorizationDecrease(
+        address operator,
+        OperatorInfo storage operatorStruct
+    ) internal {
+        uint96 totalStake = operatorStruct.tStake +
+            operatorStruct.nuInTStake +
+            operatorStruct.keepInTStake;
+        for (
+            uint256 i = 0;
+            i < operatorStruct.authorizedApplications.length;
+            i++
+        ) {
+            address application = operatorStruct.authorizedApplications[i];
+            AppAuthorization storage authorization = operatorStruct
+                .authorizations[application];
+            if (authorization.authorized <= totalStake) {
+                continue;
+            }
+
+            bool successful = true;
+            //slither-disable-next-line calls-loop
+            try
+                IApplication(application).involuntaryAuthorizationDecrease{
+                    gas: GAS_LIMIT_AUTHORIZATION_DECREASE
+                }(operator, totalStake)
+            {} catch {
+                successful = false;
+            }
+            authorization.authorized = totalStake;
+            if (authorization.deauthorizing > totalStake) {
+                authorization.deauthorizing = totalStake;
+            }
+            emit AuthorizationInvoluntaryDecreased(
+                operator,
+                application,
+                totalStake,
+                successful
+            );
+        }
+    }
+
+    /// @notice Convert amount from T to Keep and call `seize` in Keep staking contract.
+    ///         Returns remainder of slashing amount in T
+    function seizeKeep(
+        OperatorInfo storage operatorStruct,
+        address operator,
+        uint96 tAmountToSlash,
+        uint256 rewardMultiplier
+    ) internal returns (uint96) {
+        if (operatorStruct.keepInTStake == 0) {
+            return tAmountToSlash;
+        }
+
+        uint96 tPenalty;
+        if (tAmountToSlash <= operatorStruct.keepInTStake) {
+            tPenalty = tAmountToSlash;
+        } else {
+            tPenalty = operatorStruct.keepInTStake;
+        }
+
+        (uint256 keepPenalty, uint96 tRemainder) = tToKeep(tPenalty);
+        if (keepPenalty == 0) {
+            return tAmountToSlash;
+        }
+        tPenalty -= tRemainder;
+        operatorStruct.keepInTStake -= tPenalty;
```

Comment:
‼️ Instead of doing it, I wonder if it wouldn't be safer to do `operatorStruct.keepInTStake = getKeepAmountInT(operator)` *after* `keepStakingContract.seize` is called to make sure we are 100% in sync with KEEP contract and we are not entering a loop of disprepancies. This remainder makes me itchy and I am afraid of potential precision issues in `seize`. We shouldn't have problems there but better be safe than sorry...

Reply:
I had it in first versions, but this costs much more than this. We are syncing before so should be correct. Tests includes cases with remainder. I'm for keeping like now

Reply:
OK, let's just remember to ask auditors to be extra-careful about this part.